### PR TITLE
Support executing SSE on ALM SaaS

### PR DESCRIPTION
**Pull request by Roy-Lu**
Support executing SSE on ALM SaaS
JIRA Ticket: JENKINS-42295

### CODE REVIEW

File: src/main/java/com/hp/application/automation/tools/rest/RestClient.java [hunk 1]
```diff
@@ -4,13 +4,7 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
-import java.net.Authenticator;
-import java.net.HttpURLConnection;
-import java.net.InetSocketAddress;
-import java.net.PasswordAuthentication;
-import java.net.Proxy;
-import java.net.URL;
-import java.net.URLConnection;
```
Comment (YafimK) [inline src/main/java/com/hp/application/automation/tools/rest/RestClient.java]: Please avoid such changes - explicit import makes life easier to catch import issues

File: src/main/java/com/hp/application/automation/tools/sse/sdk/RunManager.java [hunk 2]
```diff
@@ -5,11 +5,17 @@
...
 import com.hp.application.automation.tools.sse.sdk.handler.PollHandlerFactory;
 import com.hp.application.automation.tools.sse.sdk.handler.RunHandler;
 import com.hp.application.automation.tools.sse.sdk.handler.RunHandlerFactory;
 
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  *
  * @author Effi Bar-She'an
```
Comment (YafimK) [inline src/main/java/com/hp/application/automation/tools/sse/sdk/RunManager.java]: Please add proper JavaDoc to this file

File: src/main/java/com/hp/application/automation/tools/results/service/AlmRestTool.java [hunk 3]
```diff
@@ -17,7 +16,7 @@
 import com.hp.application.automation.tools.sse.sdk.Logger;
 import com.hp.application.automation.tools.sse.sdk.ResourceAccessLevel;
 import com.hp.application.automation.tools.sse.sdk.Response;
-import com.hp.application.automation.tools.sse.sdk.RestAuthenticator;
+import com.hp.application.automation.tools.sse.sdk.authenticator.RestAuthenticator;
 
 public class AlmRestTool {
```
Comment (YafimK) [inline src/main/java/com/hp/application/automation/tools/results/service/AlmRestTool.java]: Please add JavaDoc

**Review by YafimK — CHANGES_REQUESTED** [review]
Please add proper JavaDoc to the classes - at least some General info  on existing classes

**Comment by YafimK** [comment]
approved for test build

**Comment by YafimK** [comment]
approved for test build

**Comment by Roy-Lu** [comment]
I think the hpebot is too strict.

**Comment by Roy-Lu** [comment]
AuthenticationTool.java#L68: Either log or rethrow this exception. rule
It is logged.

RestAuthenticator.java#L147: Remove this unused private "isAlreadyAuthenticated" method. rule
RestAuthenticator.java#L171: Remove this unused private "logLoggedInSuccessfully" method. rule
RestAuthenticatorSaas.java#L106: Remove this unused private "logLoggedInSuccessfully" method. 
These methods are used.

**Review by YafimK — APPROVED** [review]

**Comment by YafimK** [comment]
approved for test build

**Pull request merged**